### PR TITLE
Patch to enable push notifications (at least on Android) with Parse Server

### PR DESCRIPTION
**Pull request by jegesh**
In order to get the push notifications to work, I ended up having to add another Cloud Code function which sends the notification to a specific user, based on the `objectId` of the `ParseInstallation` object for that user.  For the code to work for me I had to remove any mention of the cn1 `Logger` class, since I don't have a pro account.  The test app doesn't work, and I wasn't super motivated to poke around in there and figure out what I needed to do to make it work, so I just left it as is, with failing tests but push notifications are definitely functional.  I also haven't tested things out on an iPhone yet.

### CODE REVIEW

**Comment by sidiabale** [comment]
Hi,

Glad to see that you made progress and thanks for the pull request. I'll merge it as soon as I find the time.

Any plans for push support on iOS as well? ;)

**Comment by jegesh** [comment]
I would love to be able support push notifications on ios as well.  Please give me some guidance on this, and I'll do what I can.  I must admit, though, that I have basically zero practical experience with ios.

File: cloud/main.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ Parse.Cloud.define("deleteFile", function(request, response) {
   } else {
 	  Parse.Cloud.httpRequest({
 		method: 'DELETE',
-		url: 'https://parse-parse4cn1.rhcloud.com/parse/files/' + filename,
```
Comment (sidiabale) [inline cloud/main.js]: parse4cn1 runs against openshift by default, not back4app so please revert this change otherwise tests will fail.

**Comment by jegesh** [comment]
Do you have any suggestions for how to get started on updating the iOS code for compatibility with Parse Server?

**Comment by sidiabale** [comment]
Hi,

Yes I do but unfortunately still haven't had the time to investigate
further or even merge your changes. I'll try to make a step before the
month ends.

Cheers

Op 21 nov. 2016 5:35 AM schreef "Yaakov Gesher" <notifications@github.com>:

**Comment by sidiabale** [comment]
@jegesh I've incorporated most of the changes you made by directly pulling from your master branch. That was easier for me than partially merging this PR. Thanks once more for your contributions

**Comment by jegesh** [comment]
What about support for push in iOS? Do you have a timetable, or alternatively, give me a push in the right direction?

**Comment by sidiabale** [comment]
It seems you missed the [latest release](https://github.com/sidiabale/parse4cn1/releases/tag/parse4cn1-3.1) which includes push for iOS... Check it out.

**Comment by jegesh** [comment]
I certainly will!  And thanks for all the hard work!
